### PR TITLE
Significant Refactor

### PR DESCRIPTION
**Pull request by haqa**
Significant changes
 - Split monolithic module into separate files for each major class
 - Class/Method indentation tidying 
 - Manually merge MR #11 and MR #13 (correcting for new file locations)
 - Add command-line tools to dump data, set KVM parameters and PXE boot

Minor changes
 - Ignore changes to version.py if it happens to get imported (rewritable_git_version changes the file)
 - Add eclipse/pydev project files
 - Changed to my contact details

### CODE REVIEW

**Comment by haqa** [comment]
Last commits were to fix the tests. It transpires that "gitversion" is a little fragile so I've removed it (for now?)

File: tests/TestBase.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import os.path
+import re
+parent = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+sys.path.insert(0, parent)
+import requests
+
+'''
+Created on 10 Jul 2017
+
+@author: adrian
```
Comment (mikebryant) [inline tests/TestBase.py]: We don't normally include headers like these - that's what git is for

File: tests/TestBase.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    def _requests_post(self, url, data = None, json = None, **kwargs):
+        '''
+        Check request and send response as though it was a network connection
+        '''
+        req = self.expectXML.pop(0)
+#        print len(data)
+        data = self.UUID.sub("uuid:", data)
+#        print len(data)
+#        print len(req)
```
Comment (mikebryant) [inline tests/TestBase.py]: We shouldn't be committing commented code like this

File: tests/TestBase.py (hunk 1 above)
Comment (mikebryant) [reply tests/TestBase.py]: (This comment applies to all the files)

File: wry/AMTDevice.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                else:
+                    raise NotImplementedError('The resource %r does not define a supported method for this action.' % name)
+            except:
+                impossible.append(name)
+            else:
+                output.update(resource)
+        messages = ['# Could not dump %s' % name for name in impossible]
+        if as_json:
+            return '\n'.join(messages) + '\n' + output.as_json()
+        print '\n'.join(messages)
```
Comment (mikebryant) [inline wry/AMTDevice.py]: We probably shouldn't have prints in library code

File: wry/AMTKVM.py [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+'''
+
+AMT_KVM_ENABLEMENT_MAP = {
+    2: (True, 'Enabled In Use'),
+    6: (True, 'Enabled'),
+    3: (False, 'Disabled'),
+}
+
+
+#class EnablementMap(object):
```
Comment (mikebryant) [inline wry/AMTKVM.py]: Don't commit this, if it's not ready or used?

File: wry/AMTKVM.py [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+        ports = set()
+        if self.RESOURCES['kvmRedirectionSettingData'].get('Is5900PortEnabled'):
+            ports.add(5900)
+        if self.RESOURCES['redirectionService'].get('ListenerEnabled'):
+            ports.add(16994)
+            if self.RESOURCES['tlsSettingData'].enumerate()['AMT_TLSSettingData'][0]['Enabled']:
+                ports.add(16995)
+        return ports
+#
+#    @enabled_ports.setter
```
Comment (mikebryant) [inline wry/AMTKVM.py]: Is this chunk supposed to be here? Or can it be deleted?

File: wry/wsmanResource.py [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+        Send a request to the target and return the response
+
+        '''
+        params['uuid'] = uuid.uuid4()
+        if enumerate:
+            doc = doc % params
+        else:
+            doc = doc % params
+        if self.showxml:
+            print "==== Request ===="
```
Comment (mikebryant) [inline wry/wsmanResource.py]: prints?

**Review by mikebryant — CHANGES_REQUESTED** [review]

File: wry/wsmanResource.py (hunk 6 above)
Comment (haqa) [reply wry/wsmanResource.py]: Debugging code to allow users of the library to dump the conversations. Enabled only if self.showxml is set

File: wry/AMTKVM.py (hunk 5 above)
Comment (haqa) [reply wry/AMTKVM.py]: I've deleted it

File: wry/AMTKVM.py (hunk 4 above)
Comment (haqa) [reply wry/AMTKVM.py]: It should have been removed. It has now.

File: wry/AMTDevice.py (hunk 3 above)
Comment (haqa) [reply wry/AMTDevice.py]: This a dump method who's doc-comment states it's purpose - To print that current state of AMT settings

File: tests/TestBase.py (hunk 2 above)
Comment (haqa) [reply tests/TestBase.py]: Changed to conditional on a flag (With explanatory comment)

File: tests/TestBase.py (hunk 1 above)
Comment (haqa) [reply tests/TestBase.py]: I'll remove them and update my Eclipse template for this project.

File: wry/AMTDevice.py (hunk 3 above)
Comment (haqa) [reply wry/AMTDevice.py]: Removed

**Comment by mikebryant** [comment]
cc @godley could you help me take a look at this?

**Review by mikebryant — APPROVED** [review]

**Pull request merged**